ballerina/types.bal

```diff
@@ -14,172 +14,550 @@
 // specific language governing permissions and limitations
 // under the License.
 
-import ballerinax/'client.config;
+import ballerina/time;
 
-# Represents the AWS SNS Connection Configuration.
-@display {label: "Connection Config"}
-public type ConnectionConfig record {|
-    *config:ConnectionConfig;
-    never auth?;
-    # Credentials to authenticate client 
-    AwsCredentials|AwsTemporaryCredentials credentials;
-    # Region of SNS resource
-    string region = "us-east-1";
-|};
+# The hashing algorithm used while creating the signature of the notifications, subscription confirmations, or 
+# unsubscribe confirmation messages sent by Amazon SNS.
+# 
+# + SignatureVersion1 - Amazon SNS creates the signature based on the SHA1 hash of the message
+# + SignatureVersion2 - Amazon SNS creates the signature based on the SHA256 hash of the message
+public enum SignatureVersion {
+    SignatureVersion1 = "1",
+    SignatureVersion2 = "2"
+}
 
-# AWS temporary credentials.
-#
-# + accessKeyId - Access key Id
-# + secretAccessKey - Security access key
-# + securityToken - Security token
-public type AwsTemporaryCredentials record {
-    string accessKeyId;
-    @display {
-        label: "",
-        kind: "password"
-    }
-    string secretAccessKey;
-    @display {
-        label: "",
-        kind: "password"
-    }
-    string securityToken;
-};
+# The function that Amazon SNS uses to calculate the time to wait between retries.
+public enum BackoffFunction {
+    ARITHMETIC = "arithmetic",
+    EXPONENTIAL = "exponential",
+    GEOMETRIC = "geometric",
+    LINEAR = "linear"
+}
 
-# AWS credentials.
-#
-# + accessKeyId - Access key Id
-# + secretAccessKey - Security access key
-public type AwsCredentials record {
-    string accessKeyId;
-    @display {
-        label: "",
-        kind: "password"
-    }
-    string secretAccessKey;
-};
+# The content type of the notification being sent to HTTP/S endpoints.
```

Comment:
```suggestion
# The content type of the notification being sent to HTTPS endpoints.
```

Reply:
Fixed in [020b0a4](https://github.com/ballerina-platform/module-ballerinax-aws.sns/pull/40/commits/020b0a40d31509c3e7b499cf2ea5c620bbfc9b2b)